src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -172,6 +172,36 @@ static void builderBulkDelete(std::string& bulkData, std::string_view id, std::s
     bulkData.append("\n");
 }
 
+static void builderDeleteByQuery(std::string& bulkData, const std::vector<std::string>& agentIds)
+{
+    bulkData.append(R"({
+  "query": {
+    "bool": {
+      "filter": {
+        "terms": {
+          "agent.id": [)");
+
+    // Append each agent ID to the JSON array
+    for (size_t i = 0; i < agentIds.size(); ++i)
+    {
+        bulkData.append(R"(")");
+        bulkData.append(agentIds[i]);
+        bulkData.append(R"(")");
+        if (i < agentIds.size() - 1)
+        {
+            bulkData.append(", ");
+        }
+    }
```

Comment:
To avoid conditional complexity
```suggestion
   // Append each agent ID to the JSON array
   for (const auto& agentId : agentIds)
    {
        bulkData.append(R"(")");
        bulkData.append(agentId);
        bulkData.append(R"(")");
        bulkData.append(",");
    }
    bulkData.pop_back(); // Remove the last comma
```

Reply:
Thanks, done!